From b1f85da3fa02b52913709b8433cf8ba41370a54d Mon Sep 17 00:00:00 2001
From: Alexander Forsch <alexander.forsch@bytesatwork.ch>
Date: Mon, 19 Aug 2024 15:02:27 +0200
Subject: [PATCH] dt-bindings: arm: bytesatwork: Add module and board

Add document the bytes at work AG vendor prefix.
Add byteENGINE and byteDEVKIT.

Signed-off-by: Alexander Forsch <alexander.forsch@bytesatwork.ch>
Signed-off-by: Daniel Ammann <daniel.ammann@bytesatwork.ch>
---
 .../bindings/arm/bytesatwork/bytesatwork.yaml | 30 +++++++++++++++++++
 .../devicetree/bindings/vendor-prefixes.yaml  |  2 ++
 MAINTAINERS                                   |  1 +
 3 files changed, 33 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/arm/bytesatwork/bytesatwork.yaml

diff --git a/Documentation/devicetree/bindings/arm/bytesatwork/bytesatwork.yaml b/Documentation/devicetree/bindings/arm/bytesatwork/bytesatwork.yaml
new file mode 100644
index 000000000000..567086eb9248
--- /dev/null
+++ b/Documentation/devicetree/bindings/arm/bytesatwork/bytesatwork.yaml
@@ -0,0 +1,30 @@
+# SPDX-License-Identifier: GPL-2.0-only OR BSD-2-Clause
+%YAML 1.2
+---
+$id: http://devicetree.org/schemas/arm/bytesatwork/bytesatwork.yaml#
+$schema: http://devicetree.org/meta-schemas/core.yaml#
+
+title: bytes at work Platforms Device Tree Bindings
+
+maintainers:
+  - Daniel Ammann <daniel.ammann@bytesatwork.ch>
+
+properties:
+  $nodename:
+    const: "/"
+  compatible:
+    oneOf:
+      - description: bytes at work byteENGINE
+        items:
+          - enum:
+              - batw,m5
+          - const: batw,m5
+      - description: bytes at work byteDEVKIT
+        items:
+          - enum:
+              - batw,bytedevkit
+          - const: batw,bytedevkit
+
+additionalProperties: true
+
+...
diff --git a/Documentation/devicetree/bindings/vendor-prefixes.yaml b/Documentation/devicetree/bindings/vendor-prefixes.yaml
index 51984990b839..a2ae6310a650 100644
--- a/Documentation/devicetree/bindings/vendor-prefixes.yaml
+++ b/Documentation/devicetree/bindings/vendor-prefixes.yaml
@@ -180,6 +180,8 @@ patternProperties:
     description: BAIKAL ELECTRONICS, JSC
   "^bananapi,.*":
     description: BIPAI KEJI LIMITED
+  "^batw,.*":
+    description: bytes at work AG
   "^beacon,.*":
     description: Compass Electronics Group, LLC
   "^beagle,.*":
diff --git a/MAINTAINERS b/MAINTAINERS
index 8847a58440e8..b6f9b8a8bccd 100644
--- a/MAINTAINERS
+++ b/MAINTAINERS
@@ -4502,6 +4502,7 @@ BYTES AT WORK MODULES AND BOARDS
 M:	Daniel Ammann <daniel.ammann@bytesatwork.ch>
 S:	Maintained
 F:	.github/workflows/checkpatch.yml
+F:	Documentation/devicetree/bindings/arm/bytesatwork/bytesatwork.yaml
 
 C-MEDIA CMI8788 DRIVER
 M:	Clemens Ladisch <clemens@ladisch.de>
-- 
2.39.2

